addons/account/models/account_move_line.py

```diff
@@ -556,15 +556,17 @@ def _compute_account_id(self):
                 elif line.move_id.is_purchase_document(include_receipts=True):
                     line.account_id = accounts['expense'] or line.account_id
             elif line.partner_id:
-                line.account_id = self.env['account.account']._order_by_frequency_per_partner(
+                most_frequent_account = self.env['account.account']._order_by_frequency_per_partner(
                     company_id=line.company_id.id,
                     partner_id=line.partner_id.id,
                     move_type=line.move_id.move_type,
                     limit=1,
-                )[0]
+                )
+                if most_frequent_account:
+                    line.account_id = most_frequent_account[0]
         for line in self:
             if not line.account_id and line.display_type not in ('line_section', 'line_note'):
-                line.account_id = line.company_id.account_journal_suspense_account_id
+                line.account_id = line.move_id.journal_id.default_account_id
```

Comment:
`default_account_id` is not required

Reply:
How so ? It is failing if we do not set it

Reply:
It is not required on the company, so it is not a complete fail safe.